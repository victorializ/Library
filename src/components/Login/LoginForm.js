import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import {
  Button,
  Form,
  Loader,
  Header,
  Segment
} from 'semantic-ui-react';

import { ErrorMessage } from '../../components';
import { constants } from "../../i18n";
import { colors } from "../../assets/semantic-colors";

function Login({
  loading,
  isLoggedIn,
  error, 
  login, 
  logout, 
  isAdmin
}) {
  const [ email, setEmail ] = useState('');
  const [ password, setPassword ] = useState('');

  return (
    <div className="center">
      <Segment color={colors.primary} className='login__form'>
        { loading && <Loader active inline='centered' /> }
        { error && <ErrorMessage text={error.message} /> }
        {
          !isLoggedIn ?
            <>
              <Form>
                <Form.Field>
                  <label>
                    {constants.email}
                  </label>
                  <input 
                    value={email}
                    placeholder={constants.email} 
                    onChange={({target: {value}}) => setEmail(value)}/>
                </Form.Field>
                <Form.Field>
                  <label>
                    {constants.password}
                  </label>
                  <input 
                    type='password'
                    value={password}
                    placeholder={constants.password} 
                    onChange={({target: {value}}) => setPassword(value)}/>
                </Form.Field>
                <Button 
                  attached
                  onClick={() => login(email, password)}
                  disabled={!email || !password}>
                  {constants.login}
                </Button>
              </Form>
              <Link to='/registration'>{constants.register}</Link>
            </> 
            :
            <>
              <Header textAlign="center">
                {constants.loggedin}
              </Header>
              <Button.Group>
                {isAdmin ? 
                  <Button 
                    as={Link} to='/admin' 
                    color={colors.red}>
                    {constants.admin}
                  </Button>
                  :
                  <Button 
                    as={Link} to='/orders' 
                    color={colors.primary}>
                    {constants.personalAccount}
                  </Button>
                }
                <Button.Or />
                <Button onClick={logout}>
                  {constants.logout}
              </Button>
              </Button.Group>
            </>
        }
      </Segment>
    </div>
  );
}

export { Login as LoginForm };