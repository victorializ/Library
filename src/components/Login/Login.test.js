import React from 'react';
import { shallow } from 'enzyme';

import { LoginForm } from './LoginForm';

describe('Login form component', () => {
    it('renders correctly', () => {
        expect(shallow(<LoginForm />)).toMatchSnapshot();
    }); 
    it('renders loader when loading', () => {
        const wrapper = shallow(<LoginForm loading={true} />);
        expect(wrapper.find('.login__loader').exists()).toBeTruthy();
    });
    it('renders error when error', () => {
        const error = {message: 'This is error!'};
        const wrapper = shallow(<LoginForm error={error} />);
        expect(wrapper.find('.login__error').exists()).toBeTruthy();
    });
    it('renders link to profile when user is logged in', () => {
        const wrapper = shallow(<LoginForm isLoggedIn={true} />);
        expect(wrapper.find('.logged-in').exists()).toBeTruthy();
    });
    it('not renders link to profile when user is not logged in', () => {
        const wrapper = shallow(<LoginForm isLoggedIn={false} />);
        expect(wrapper.find('.logged-in').exists()).toBeFalsy();
    });
    it('updates user email on change', () => {
        const wrapper = shallow(<LoginForm />);
        const selector = '.login__input-email';
        const event = {target: {value: 'text'}};
        wrapper.find(selector).simulate('change', event);
        expect(wrapper.find(selector).prop('value')).toEqual(event.target.value);
    });
    it('updates user password on change', () => {
        const wrapper = shallow(<LoginForm />);
        const selector = '.login__input-password';
        const event = {target: {value: 'text'}};
        wrapper.find(selector).simulate('change', event);
        expect(wrapper.find(selector).prop('value')).toEqual(event.target.value);
    });
    it('calles login function', () => {
        const login = jest.fn();
        const wrapper = shallow(<LoginForm login={login} />);
        const selector = '.login__button';
        wrapper.find(selector).simulate('click');
        expect(login.mock.calls.length).toBe(1);
    });
});