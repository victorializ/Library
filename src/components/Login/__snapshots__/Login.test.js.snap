// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component renders correctly 1`] = `
<div
  className="center"
>
  <Segment
    className="login__form"
    color="olive"
  >
    <Form
      as="form"
    >
      <FormField>
        <label>
          Email
        </label>
        <input
          className="login__input-email"
          onChange={[Function]}
          placeholder="Email"
          value=""
        />
      </FormField>
      <FormField>
        <label>
          Password
        </label>
        <input
          className="login__input-password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </FormField>
      <Button
        as="button"
        attached={true}
        className="login__button"
        disabled={true}
        onClick={[Function]}
      >
        Login
      </Button>
    </Form>
    <Link
      className="registration__link"
      to="/registration"
    >
      Registration
    </Link>
  </Segment>
</div>
`;
